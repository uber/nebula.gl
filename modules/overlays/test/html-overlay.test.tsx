import React from 'react';
/*
import Renderer from 'react-test-renderer';

import DeckGL from '@deck.gl/react';
import { createHeadlessContext } from '@luma.gl/test-utils';
*/
import HtmlOverlay from '../src/html-overlay';
import HtmlOverlayItem from '../src/html-overlay-item';

/*
const initialViewState = {
  longitude: 0,
  latitude: 0,
  zoom: 10,
  pitch: 0,
  bearing: 0,
};

it('test HtmlOverlay map center', () => {
  const Component = (
    // @ts-ignore
    <DeckGL
      gl={createHeadlessContext({ width: 100, height: 100, webgl1: true })}
      initialViewState={initialViewState}
    >
      <HtmlOverlay>
        <HtmlOverlayItem coordinates={[0, 0, 0]}>Map Center Zero Elevation</HtmlOverlayItem>
        <HtmlOverlayItem coordinates={[0, 0, 50000]}>Map Center 50km Elevation</HtmlOverlayItem>
      </HtmlOverlay>
    </DeckGL>
  );
  const renderer = Renderer.create(Component);
  expect(renderer.toJSON()).toMatchSnapshot();
});
*/

it('HtmlOverlay is able to handle a single null child', () => {
  const layer = new HtmlOverlay({
    viewport: { project: (coords) => coords },
  });

  expect(layer.render()).toMatchSnapshot();
});

it('HtmlOverlay is able to handle a null child with other valid children', () => {
  const children = [
    <HtmlOverlayItem key={1} coordinates={[0, 0, 0]}>
      Map Center Zero Elevation
    </HtmlOverlayItem>,
    <HtmlOverlayItem key={2} coordinates={[0, 0, 50000]}>
      Map Center 50km Elevation
    </HtmlOverlayItem>,
    null,
  ];

  const layer = new HtmlOverlay({
    children,
    viewport: { project: (coords) => coords },
  });
  expect(layer.render()).toMatchSnapshot();
});

it('HtmlOverlay is able to handle no children', () => {
  const layer = new HtmlOverlay({ viewport: { project: (coords) => coords } });
  expect(layer.render()).toMatchSnapshot();
});
