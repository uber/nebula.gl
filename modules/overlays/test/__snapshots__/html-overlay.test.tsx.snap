// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlOverlay is able to handle a null child with other valid children 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "hidden",
      "pointerEvents": "none",
      "position": "absolute",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <HtmlOverlayItem
    coordinates={
      Array [
        0,
        0,
        0,
      ]
    }
    x={0}
    y={0}
  >
    Map Center Zero Elevation
  </HtmlOverlayItem>
  <HtmlOverlayItem
    coordinates={
      Array [
        0,
        0,
        50000,
      ]
    }
    x={0}
    y={0}
  >
    Map Center 50km Elevation
  </HtmlOverlayItem>
</div>
`;

exports[`HtmlOverlay is able to handle a single null child 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "hidden",
      "pointerEvents": "none",
      "position": "absolute",
      "width": "100%",
      "zIndex": 1,
    }
  }
/>
`;

exports[`HtmlOverlay is able to handle no children 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "hidden",
      "pointerEvents": "none",
      "position": "absolute",
      "width": "100%",
      "zIndex": 1,
    }
  }
/>
`;
